website/source/docs/commands/namespace/update.html.md.erb

```diff
@@ -0,0 +1,95 @@
+---
+layout: "docs"
+page_title: "Commands: Namespace Update"
+sidebar_current: "docs-commands-namespace-update"
+---
+
+<%= enterprise_alert :consul %>
+
+# Consul Namespace Update
+
+Command: `consul namespace update`
+
+This `namespace update` command updates a namespaces using the CLI parameters provided. 
+This was added in Consul Enterprise 1.7.2.
+
+## Usage
+
+Usage: `consul namespace update -name <namespace name> [options]`
+
+Request a namespace to be update. Construction of the namespace definition is handled by this command
+from the CLI arguments. Some parts of the Namespace such as ACL configurations and meta can be merged
+with the existing namespace definition.
+
+#### API Options
+
+<%= partial "docs/commands/http_api_options_client" %>
+<%= partial "docs/commands/http_api_options_server" %>
+
+#### Command Options
+
+* `-default-policy-id=<value>` - ID of a policy from the default namespace to inject for all tokens
+   in this namespace. May be specified multiple times.
+
+* `-default-policy-name=<value>` - Name of a policy from the default namespace to inject for all
+   tokens in this namespace. May be specified multiple times.
+
+* `-default-role-id=<value>` - ID of a role from the default namespace to inject for all tokens in
+   this namespace. May be specified multiple times.
+
+* `-default-role-name=<value>` - Name of a role from the default namespace to inject for all tokens
+   in this namespace. May be specified multiple times.
+
+* `-description=<string>` - A description of the namespace.
+
+* `-format=<string>` - How to output the results. The choices are: pretty or json
+
+* `-merge-acls` - Merge the new ACL policies and roles with the existing values.
+
+* `-merge-meta` - Merge new meta values with existing meta.
+
+* `-meta=<value>` - Metadata to set on the namespace, formatted as key=value. This flag
+   may be specified multiple times to set multiple meta fields
+
+* `-name=<string>` - The namespace's name. This flag is required.
+
+* `-show-meta` - Indicates that namespace metadata such as the raft indices should
+   be shown for the namespace
+
+## Examples
+
+Update a namespace with a new description:
+
+```sh
+$ consul namespace update -name "team-1" -description "example description"
+Name: team-1
+Description:
+   example description
+```
+
+Showing Raft Metadata:
+
+```sh
+$ consul namespace update -name team-1 -show-meta -default-policy-id 1206bf1c-6239-46e8-b9f8-b426667cf428
+Name: team-1
+ACLs:
+   Default Policies:
+      1206bf1c-6239-46e8-b9f8-b426667cf428 / team1-universal-policy
+Create Index: 339
+Modify Index: 344
+```
+
+JSON Format:
+
+```sh
+$ consul namespace update -name foo -description "Example Namespace" -meta "team-id=574407f3-8b26-4c84-8e51-028bb8cbdd37" -format=json
```

Comment:
Does the team id have to be a UUID? If its just a string, could use a more descriptive example like `team-id=web`

Reply:
meta is free-form key/value data. This was long but when thinking about the example I was going through my mind thinking about what sorts of kv data I would want to associate with the namespace. One would be a uuid or some other "group" id from an external Auth system for mapping users to namespaces.

Maybe that overcomplicates things here? Given that it can be almost anything, would you have something else you would prefer to see here?

Reply:
I could make it really arbitrary and set `-meta team-color=pink` or something like that.

Reply:
I changed it to `-meta external-source=kubernetes`. This is one of the two meta values consul-k8s will set on namespaces and consul-k8s was the original requestor of meta on namespaces.